pennylane/measurements.py

```diff
@@ -587,3 +593,115 @@ def circuit():
     """
     # pylint: disable=protected-access
     return MeasurementProcess(State, wires=qml.wires.Wires(wires))
+
+
+T = TypeVar("T")
+
+
+class MeasurementValueError(ValueError):
+    """Error raised when an unknown measurement value is being used."""
+
+
+class MeasurementValue(Generic[T]):
+    """A class representing unknown measurement outcomes in the qubit model.
+
+    By default, measurements in the computational basis are assumed.
+
+    Args:
+        measurement_id (str): The id of the measurement that this object depends on.
+        zero_case (float): the measurement outcome in case the first
+            measurement outcome was obtained
+        one_case (float): the measurement outcome in case the second
+            measurement outcome was obtained
```

Comment:
Out of curiosity, how would we generalize this to a multiqubit `MeasureValue`?

Reply:
Good question! Likely we'd need to accept a list of cases, instead of having 2 of them baked in. This should not be a major issue though, as the `MeasurementValue` class is developer facing.